### PR TITLE
changes/fixes to metadata.cloud.* fields collected for GCP

### PR DESCRIPTION
**Pull request by trentm**
- cloud.instance.id must be a string
- change cloud.project.id to be the 'projectId' (per spec change)
- drop cloud.project.name
- fix cloud.machine.type

Closes: https://github.com/elastic/apm-agent-ruby/issues/1414

### CODE REVIEW

File: lib/elastic_apm/metadata/cloud_info.rb [hunk 1]
```diff
@@ -97,13 +97,12 @@ def fetch_gcp
         zone = metadata["instance"]["zone"]&.split("/")&.at(-1)
 
         self.provider = "gcp"
-        self.instance_id = metadata["instance"]["id"]
+        self.instance_id = metadata["instance"]["id"].to_s
```
Comment (estolfo) [inline lib/elastic_apm/metadata/cloud_info.rb:100]: The [issue description](https://github.com/elastic/apm/issues/826) says to check that the int is being handled correctly but doesn't say it should be collected as a string. Is using a string the solution?

File: lib/elastic_apm/metadata/cloud_info.rb [hunk 2]
```diff
@@ -97,13 +97,12 @@ def fetch_gcp
...
         self.instance_name = metadata["instance"]["name"]
-        self.project_id = metadata["project"]["numericProjectId"]
-        self.project_name = metadata["project"]["projectId"]
+        self.project_id = metadata["project"]["projectId"]
         self.availability_zone = zone
         self.region = zone.split("-")[0..-2].join("-")
-        self.machine_type = metadata["instance"]["machineType"]
+        self.machine_type = metadata["instance"]["machineType"].split("/")&.at(-1)
```
Comment (estolfo) [inline lib/elastic_apm/metadata/cloud_info.rb:105]:
```suggestion
        self.machine_type = metadata["instance"]["machineType"].split("/")[-1]
```

**Review by estolfo — CHANGES_REQUESTED** [review]
Thanks for doing this for the Ruby agent!

File: lib/elastic_apm/metadata/cloud_info.rb (hunk 1 above)
Comment (trentm) [reply lib/elastic_apm/metadata/cloud_info.rb:100]: This was a bug in a the Ruby impl unrelated to the spec change for https://github.com/elastic/apm/issues/826.

The spec had already said:

> `instance.id` as a string

and the intake schema requires a string (https://github.com/elastic/apm-agent-ruby/blob/8d4a30cd3f592d6fcb7fc97826d71ec6d908b4b7/spec/fixtures/metadata.json#L52-L57).

**Review by estolfo — APPROVED** [review]
Thanks!

**Comment by estolfo** [comment]
@trentm can you rebase against main? I've just merged in a fix for the test failure on ruby 2.4.

**Comment by trentm** [comment]
> @trentm can you rebase against main? I've just merged in a fix for the test failure on ruby 2.4.

Done.

**Pull request merged**